minecraft-waterfall/minecraft-waterfall.json

```diff
@@ -1,86 +1,95 @@
+      "display": "build",
       "required": true,
-      "desc": "\"java\" for system standard, else direct path to java",
-      "display": "path to java executable",
-      "internal": false,
+      "value": "477"
+    },
+    "ip": {
       "type": "string",
-      "userEdit": true
+      "desc": "What IP to bind the server to",
+      "display": "IP",
+      "required": true,
+      "value": "0.0.0.0"
     },
-    "version": {
-      "value": "1.16",
+    "java": {
+      "type": "string",
+      "desc": "\"java\" for system standard, else direct path to java",
+      "display": "path to java executable",
       "required": true,
-      "desc": "Version of Minecraft to install. Must be specified as a <strong>minor release number</strong>, e.g. 1.16",
-      "display": "Version",
-      "internal": false
+      "value": "java",
+      "userEdit": true
     },
     "memory": {
-      "value": "512",
-      "required": true,
+      "type": "integer",
       "desc": "How much memory in MB to allocate to the Waterfall server. It is generally recommended to allocated 512MB RAM for every 500 players. Please keep in mind that additional plugins may impact the amount needed in your case.",
       "display": "Memory (MB)",
-      "internal": false,
-      "type": "integer"
+      "required": true,
+      "value": "512"
     },
-    "ip": {
-      "value": "0.0.0.0",
+    "motd": {
+      "type": "string",
+      "desc": "This is the message that is displayed in the server list of the client, below the name. The MOTD does support <a href='https://minecraft.gamepedia.com/Formatting_codes' target='_blank'>color and formatting codes</a>.",
+      "display": "MOTD message of the day",
       "required": true,
-      "desc": "What IP to bind the server to",
-      "display": "IP",
-      "internal": false
+      "value": "A Waterfall Server &9hosted on PufferPanel"
     },
     "port": {
-      "value": "25577",
-      "required": true,
+      "type": "integer",
       "desc": "What port to bind the server to",
       "display": "Port",
-      "internal": false,
-      "type": "integer"
+      "required": true,
+      "value": "25577"
     },
-    "motd": {
-      "value": "A Waterfall Server &9hosted on PufferPanel",
+    "version": {
+      "type": "string",
+      "desc": "Version of Minecraft to install. Must be specified as a <strong>minor release number</strong>, e.g. 1.16",
+      "display": "Version",
       "required": true,
-      "desc": "This is the message that is displayed in the server list of the client, below the name. The MOTD does support <a href='https://minecraft.gamepedia.com/Formatting_codes' target='_blank'>color and formatting codes</a>.",
-      "display": "MOTD message of the day",
-      "internal": false
+      "value": "1.18"
+    }
+  },
+  "environment": {
+    "type": "standard"
+  },
+  "supportedEnvironments": [
+    {
+      "type": "standard"
+    },
+    {
+      "image": "openjdk:8-alpine",
```

Comment:
Doesn't 1.18 need java 16 or something higher than 8?

Reply:
Waterfall gets built with java 17 and java 8, so it should run with java 8+

Reply:
https://github.com/PaperMC/Waterfall/blob/master/.github/workflows/build.yml